comp/core/workloadmeta/merge.go

```diff
@@ -83,7 +83,9 @@ func mergeContainerPort(mergeMap map[string]ContainerPort, port ContainerPort) {
 	existingPort, found := mergeMap[portKey]
 
 	if found {
-		if existingPort.Name == "" && port.Name != "" {
+		if (existingPort.Name == "" && port.Name != "") ||
+			(existingPort.HostIP == "" && port.HostIP != "") ||
+			(existingPort.HostPort == 0 && port.HostPort != 0) {
```

Comment:
could you add some unittest to test the new merge logic

Reply:
https://github.com/DataDog/datadog-agent/pull/23250/commits/23962523ff396d41ac60bef74695adfba4b942b7